bundles/http/Citrix/Canvas-LMS/script.js

```diff
@@ -0,0 +1,950 @@
+const moment = library.load('moment-timezone');
+
+const startDate = moment.utc().subtract(30, 'd').format();
+const endDate = moment.utc().add(30, 'd').format();
+const now = moment.utc().format();
+const end = moment.utc().add(1, 'h').format();
+const quarter = moment(now).quarter();
+const perPage = 100;
+
+async function canvasFullSync({ dataStore, client, latestSynchronizationTime = null, integrationParameters }) {
+	let courses = [];
+	let enrollmentIncrement = 1;
+	let newPageExist;
+
+	do {
+		newPageExist = false;
+		const enrollmentTermResponse = await client.fetch(
+			`/api/v1/accounts/${integrationParameters.accountId}/terms?page=${enrollmentIncrement}&per_page=${perPage}`
+		);
+		if (!enrollmentTermResponse.ok) {
+			throw new Error(
+				`Enrollment Term sync failed ${enrollmentTermResponse.status}:${enrollmentTermResponse.statusText}.`
+			);
+		}
+		const tempEnrollmentLink = enrollmentTermResponse.headers.map.link.split(',');
+		const enrollmentTerm = await enrollmentTermResponse.json();
+		const filteredValue = enrollmentTerm.enrollment_terms.filter((value) => {
+			if (
+				(value.start_at && value.end_at) &&
+				moment(value.start_at).isBetween(
+					moment().quarter(quarter).startOf('quarter'),
+					moment().quarter(quarter).endOf('quarter')
+				) ||
+				moment(value.end_at).isBetween(
+					moment().quarter(quarter).startOf('quarter'),
+					moment().quarter(quarter).endOf('quarter')
+				)
+			)
+				return true;
+		});
+		for (const enrollmentTermValue of filteredValue) {
+			let courseIncrement = 1;
+			let nextPageExists;
+
+			do {
+				nextPageExists = false;
+				const courseResponse = await client.fetch(
+					`api/v1/accounts/${integrationParameters.subAccountId}/courses?page=${courseIncrement}&per_page=${perPage}&enrollment_term_id=${enrollmentTermValue.id}`
+				);
+				courseIncrement++;
+				if (!courseResponse.ok) {
+					throw new Error(`Courses sync failed ${courseResponse.status}:${courseResponse.statusText}.`);
+				}
+				const tempCourseLink = courseResponse.headers.map.link.split(',');
+				const course = await courseResponse.json();
+
+				for (const courseValue of course) {
+					let courseDataDetails = {
+						id: courseValue.id,
+						course_code: courseValue.course_code,
+						name: courseValue.name
+					};
+					courses.push(courseDataDetails);
+					dataStore.save('Courses', courseDataDetails);
+				}
+				tempCourseLink.forEach((element) => {
+					const courseLink = element.split('; ')[1];
+					if (courseLink === `rel="next"`) {
+						courseIncrement = element
+						.split('; ')[0]
+						.replace(/^<+|>+$/g, '')
+						.split('&page=')[1]
+						.split('&per_page')[0];
+						nextPageExists = true;
+					}
```

Comment:
Let's put this code into a function and call the function where ever required to reduce the redundancy in the code.

Reply:
We have achieved code redundancy by creating the function